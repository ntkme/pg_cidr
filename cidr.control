# cidr extension
comment = 'transform ip range to cidr'
default_version = '1.0'
relocatable = true
