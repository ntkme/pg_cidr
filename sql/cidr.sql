CREATE EXTENSION cidr;

SELECT cidr(inet '1.1.1.0', inet '1.1.1.255');

SELECT cidr(inet '1.1.1.0', inet '1.1.1.8');

SELECT cidr(inet '8.8.8.8', inet '8.255.255.255');

SELECT cidr(inet '0.0.0.0', inet '255.255.255.255');

SELECT cidr(inet '1::', inet '1:ffff:ffff:ffff:ffff:ffff:ffff:ffff');

SELECT cidr(inet '1::', inet '2::');

SELECT cidr(inet '2001:4860:4860:0000:0000:0000:0000:8888', inet '2001:4860:ffff:ffff:ffff:ffff:ffff:ffff');

SELECT cidr(inet '::', inet 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff');

SELECT cidr(inet '::', inet '1.1.0.0');

SELECT cidr(inet '1.1.1.0/24', inet '1.1.0.0/16');

SELECT cidr(inet '1.1.1.0', inet '1.1.0.0');
