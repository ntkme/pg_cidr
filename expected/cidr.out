CREATE EXTENSION cidr;
SELECT cidr(inet '1.1.1.0', inet '1.1.1.255');
    cidr    
------------
 1.1.1.0/24
(1 row)

SELECT cidr(inet '1.1.1.0', inet '1.1.1.8');
    cidr    
------------
 1.1.1.0/29
 1.1.1.8/32
(2 rows)

SELECT cidr(inet '8.8.8.8', inet '8.255.255.255');
     cidr     
--------------
 8.8.8.8/29
 8.8.8.16/28
 8.8.8.32/27
 8.8.8.64/26
 8.8.8.128/25
 8.8.9.0/24
 8.8.10.0/23
 8.8.12.0/22
 8.8.16.0/20
 8.8.32.0/19
 8.8.64.0/18
 8.8.128.0/17
 8.9.0.0/16
 8.10.0.0/15
 8.12.0.0/14
 8.16.0.0/12
 8.32.0.0/11
 8.64.0.0/10
 8.128.0.0/9
(19 rows)

SELECT cidr(inet '0.0.0.0', inet '255.255.255.255');
   cidr    
-----------
 0.0.0.0/0
(1 row)

SELECT cidr(inet '1::', inet '1:ffff:ffff:ffff:ffff:ffff:ffff:ffff');
  cidr  
--------
 1::/16
(1 row)

SELECT cidr(inet '1::', inet '2::');
  cidr   
---------
 1::/16
 2::/128
(2 rows)

SELECT cidr(inet '2001:4860:4860:0000:0000:0000:0000:8888', inet '2001:4860:ffff:ffff:ffff:ffff:ffff:ffff');
            cidr             
-----------------------------
 2001:4860:4860::8888/125
 2001:4860:4860::8890/124
 2001:4860:4860::88a0/123
 2001:4860:4860::88c0/122
 2001:4860:4860::8900/120
 2001:4860:4860::8a00/119
 2001:4860:4860::8c00/118
 2001:4860:4860::9000/116
 2001:4860:4860::a000/115
 2001:4860:4860::c000/114
 2001:4860:4860::1:0/112
 2001:4860:4860::2:0/111
 2001:4860:4860::4:0/110
 2001:4860:4860::8:0/109
 2001:4860:4860::10:0/108
 2001:4860:4860::20:0/107
 2001:4860:4860::40:0/106
 2001:4860:4860::80:0/105
 2001:4860:4860::100:0/104
 2001:4860:4860::200:0/103
 2001:4860:4860::400:0/102
 2001:4860:4860::800:0/101
 2001:4860:4860::1000:0/100
 2001:4860:4860::2000:0/99
 2001:4860:4860::4000:0/98
 2001:4860:4860::8000:0/97
 2001:4860:4860::1:0:0/96
 2001:4860:4860::2:0:0/95
 2001:4860:4860::4:0:0/94
 2001:4860:4860::8:0:0/93
 2001:4860:4860::10:0:0/92
 2001:4860:4860::20:0:0/91
 2001:4860:4860::40:0:0/90
 2001:4860:4860::80:0:0/89
 2001:4860:4860::100:0:0/88
 2001:4860:4860::200:0:0/87
 2001:4860:4860::400:0:0/86
 2001:4860:4860::800:0:0/85
 2001:4860:4860::1000:0:0/84
 2001:4860:4860::2000:0:0/83
 2001:4860:4860::4000:0:0/82
 2001:4860:4860::8000:0:0/81
 2001:4860:4860:0:1::/80
 2001:4860:4860:0:2::/79
 2001:4860:4860:0:4::/78
 2001:4860:4860:0:8::/77
 2001:4860:4860:0:10::/76
 2001:4860:4860:0:20::/75
 2001:4860:4860:0:40::/74
 2001:4860:4860:0:80::/73
 2001:4860:4860:0:100::/72
 2001:4860:4860:0:200::/71
 2001:4860:4860:0:400::/70
 2001:4860:4860:0:800::/69
 2001:4860:4860:0:1000::/68
 2001:4860:4860:0:2000::/67
 2001:4860:4860:0:4000::/66
 2001:4860:4860:0:8000::/65
 2001:4860:4860:1::/64
 2001:4860:4860:2::/63
 2001:4860:4860:4::/62
 2001:4860:4860:8::/61
 2001:4860:4860:10::/60
 2001:4860:4860:20::/59
 2001:4860:4860:40::/58
 2001:4860:4860:80::/57
 2001:4860:4860:100::/56
 2001:4860:4860:200::/55
 2001:4860:4860:400::/54
 2001:4860:4860:800::/53
 2001:4860:4860:1000::/52
 2001:4860:4860:2000::/51
 2001:4860:4860:4000::/50
 2001:4860:4860:8000::/49
 2001:4860:4861::/48
 2001:4860:4862::/47
 2001:4860:4864::/46
 2001:4860:4868::/45
 2001:4860:4870::/44
 2001:4860:4880::/41
 2001:4860:4900::/40
 2001:4860:4a00::/39
 2001:4860:4c00::/38
 2001:4860:5000::/36
 2001:4860:6000::/35
 2001:4860:8000::/33
(86 rows)

SELECT cidr(inet '::', inet 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff');
 cidr 
------
 ::/0
(1 row)

SELECT cidr(inet '::', inet '1.1.0.0');
ERROR:  cannot merge addresses from different families
CONTEXT:  PL/pgSQL function cidr(inet,inet) line 9 at RAISE
SELECT cidr(inet '1.1.1.0/24', inet '1.1.0.0/16');
ERROR:  cannot merge networks
CONTEXT:  PL/pgSQL function cidr(inet,inet) line 13 at RAISE
SELECT cidr(inet '1.1.1.0', inet '1.1.0.0');
ERROR:  result is out of range
CONTEXT:  PL/pgSQL function cidr(inet,inet) line 17 at RAISE
